datatypes/src/collections/feature_collection.rs

```diff
@@ -558,6 +570,94 @@ where
         ))
     }
 
+    /// Filter a column by one or more ranges.
+    /// If `keep_nulls` is false, then all nulls will be discarded.
+    pub fn column_range_filter<R>(
+        &self,
+        column: &str,
+        ranges: &[R],
+        keep_nulls: bool,
+    ) -> Result<Self>
+    where
+        R: RangeBounds<FeatureDataValue>,
+    {
+        let column_type = self.types.get(column);
+        ensure!(
+            column_type.is_some(),
+            error::ColumnDoesNotExist {
+                name: column.to_string()
+            }
+        );
+
+        let column = self
+            .table
+            .column_by_name(column)
+            .expect("checked by ensure");
+        let column_type = column_type.expect("checked by ensure");
+
+        let mut filter_array = None;
+
+        match column_type {
+            FeatureDataType::Number | FeatureDataType::NullableNumber => {
+                apply_filters(
+                    as_primitive_array::<Float64Type>(column),
+                    &mut filter_array,
+                    ranges,
+                    arrow::compute::gt_eq_scalar,
+                    arrow::compute::gt_scalar,
+                    arrow::compute::lt_eq_scalar,
+                    arrow::compute::lt_scalar,
```

Comment:
the last 4 arguments to the function are always the same. consider wrapping with a function that sets these